File: leetsync/2387-partition-array-such-that-maximum-difference-is-k/partition-array-such-that-maximum-difference-is-k.go
func partitionArray(nums []int, k int) int {
	n := len(nums)
	count := 1
	sort.Ints(nums)
	min := nums[0]
	for i := 0; i < n; i++ {
		if (nums[i] - min) > k {
			count++
			min = nums[i]
		}
	}
	return count
}

File: leetsync/2387-partition-array-such-that-maximum-difference-is-k/partition-array-such-that-maximum-difference-is-k.java
class Solution {
    public int partitionArray(int[] nums, int k) {
        int n = nums.length;
        int count = 1;
        Arrays.sort(nums);
        int min = nums[0];
        for (int i = 0; i < n; i++) {
            if ((nums[i] - min) > k) {
                min = nums[i];
                count++;
            }
        }
        return count;
    }
}